### PR TITLE
Modified indexing/query strategy and added support for Property Paths on filter query

### LINKED ISSUE #10: Link instances to their schemas / domains / organizations
The instance metadata is indexed as literal values. The instance should instead be linked to the appropriate related resources (schema / domain / organization).

### LINKED ISSUE #11: Link schemas to their domain / organization
The schema metadata is indexed as literal values. The schema should instead be linked to the appropriate related resources (domain / organization).

### LINKED ISSUE #12: Link domains to their organization
The doman metadata is indexed as literal values. The domain should instead be linked to the appropriate related resources (organization).

### LINKED ISSUE #62: Add support for property path on the filter query parameter
The [filter expression](https://bbp-nexus.epfl.ch/dev/docs/kg/api-reference/operating-on-resources.html#filter-expressions) supports URIs as a predicate, but it doesn't support navigting through predicates - called [property paths](https://www.w3.org/TR/sparql11-property-paths/)-.

Add support for it, as described in the spec.

### PR DESCRIPTION
**Pull request by umbreak**
Modified indexing and query strategy, from literals to ids.
Modified queries to support the change on the indexing strategy.
Added support for property paths on filter query parameter.

Fixes #62
Fixes #10 
Fixes #11 
Fixes #12

### CODE REVIEW

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import scala.util.Try
+
+sealed trait PropPath extends Product with Serializable
+
+object PropPath extends PropPathBuilder {
+
+  /**
+    * Path property which represents a URI or a prefixed name
+    *
+    * @param value the URI or a prefixed name.
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala]: fix param docs in the file

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/jsonld/UriJsonLDSupport.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.kg.indexing.IndexingVocab.JsonLDKeys._
+import io.circe.Json
+
+/**
+  * Typeclass definition that adds JSON-LD support into [[Uri]]
+  *
+  */
+trait UriJsonLDSupport {
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/jsonld/UriJsonLDSupport.scala]: you can just create an implicit class to turn a Uri into a Json representation with an `@id` key.

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala [hunk 3]
```diff
@@ -25,10 +24,17 @@ import ch.epfl.bluebrain.nexus.kg.indexing.{ConfiguredQualifier, Qualifier}
   * @tparam F  the monadic effect type
   * @tparam Id the generic type which defines the response's payload
   */
-class FilterQueries[F[_], Id](queryClient: SparqlQuery[F], querySettings: QuerySettings) {
-
+class FilterQueries[F[_], Id](queryClient: SparqlQuery[F], querySettings: QuerySettings)(
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala]: use the most specific filter only if it includes the less specific ones.

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala (hunk 1 above)
Comment (umbreak) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala]: Done

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/jsonld/UriJsonLDSupport.scala (hunk 2 above)
Comment (umbreak) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/jsonld/UriJsonLDSupport.scala]: Done

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala (hunk 3 above)
Comment (umbreak) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala]: Done

**Pull request merged**